packages/edge-gateway/src/gateway.js

```diff
@@ -202,8 +204,24 @@ async function getFromDotstorage (request, cid, options = {}) {
           response,
           resolutionIdentifier: host
         }
+      }),
+      ...env.cdnGateways.map(async (host) => {
```

Comment:
This was not added as a [GatewayRacer](https://github.com/web3-storage/reads/tree/main/packages/ipfs-gateway-race#usage) instance yet, but should be added in follow up.

At the moment, we have as part of this step (dotstorage cdn layer) performing requests to only if cached gateways (nftstorage.link asks w3s.link cache, and other way around). This means part of this race requires different headers, as well as non path resolution.

To use gateway racer now, we would need to add support there to specify headers per gateway target and to support path+subdomain resolution. This might not be needed, because we might go with [CF Cache Reserve + custom hostname for nftstorage.link](https://filecoinproject.slack.com/archives/C02BZPRS9HP/p1665762610428739) which would mean the onlyIfCachedHosts would not be needed anymore.